### ISSUE TITLE
Bug: `-fno_objc_arc` not passed to build commands with BwX

### ISSUE THREAD

**Issue by erikkerber** [MobileNativeFoundation/rules_xcodeproj#2206]
### Description

We have an ObjC file that uses `-fno_objc_arc`, which stopped building after 1.4.0. Likely with https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/2016

These flags don't appear to be included in the params files.

### Reproduction steps

1. Write ObjC that uses NSAutoReleasePool
2. In the objc_library rule, pass `-fno_objc_arc`
3. Generate project with BwX

It will not build as it is being built with ARC. The `-fno_objc_arc` is not passed to the build command either directly or in the params file.

### Expected behavior

`-fno_objc_arc` is propegated.

### rules_xcodeproj version

1.5.1

### Xcode version

14.3

### Bazel version

6.2

**Comment by brentleyjones** [comment]
https://github.com/MobileNativeFoundation/rules_xcodeproj/blob/main/tools/generators/legacy/src/Generator/AddTargets.swift#L438 is still here, so you should use `non_arc_srcs` for this to get it to be picked up.

We could go with your solution in #2207, but then we would get doubled up flags for people that use `non_arc_srcs`. Probably fine though?

**Comment by erikkerber** [comment]
Doubling up is fine, I noticed they were doubled up before when debugging. 

Couldn't figure out how that property is derived. Is that line above even needed if non-arc objc files pass `-fno_objc_arc` in copts?

**Comment by brentleyjones** [comment]
Yeah, that line is needed, because if one uses `non_arc_arcs` and `srcs` in the same library, then different files will have different flags, and we only grab a single file to sample the flags. So actually, your fix would break if I sample a `non_arc_srcs` file and use it as the baseline for all srcs.

**Comment by brentleyjones** [comment]
> The per-file situation would win-out here though right? (Maybe?)

No. Say you had this:

```starlark
objc_library(
    name = "lib",
    non_arc_srcs = ["a.c"],
    srcs = ["b.c"],
)
```

And say that in `optz.bzl` we pick up `a.c` as the prototype for `lib.rules_xcodeproj.c.compile.params`. If means that both `a.c` and `b.c` would compile with `-fno_objc_arc` (and `a.c` would have double flags). We don't set the `-fobjc_arc` flag on the `srcs` files (as it's implied).

**Comment by erikkerber** [comment]
Oh, I forgot that was an attribute. Why don't I fix this on our side — it's a single-file library.

**Comment by brentleyjones** [comment]
It's going to annoy me that this will break indexing in BwB mode for others out there that did the same thing. So I would still like to find a fix in the long term.
